src/main/java/com/glencoesoftware/omero/ms/image/region/ImageDataRequestHandler.java

```diff
@@ -0,0 +1,708 @@
+import ome.model.display.ChannelBinding;
+import ome.model.display.RenderingDef;
+import ome.model.enums.Family;
+import ome.model.enums.RenderingModel;
+import ome.model.units.BigResult;
+import omero.model.Length;
+import omero.model.LengthI;
+import omero.model.StatsInfo;
+import omeis.providers.re.Renderer;
+import omeis.providers.re.codomain.CodomainChain;
+import omeis.providers.re.codomain.CodomainMapContext;
+import omeis.providers.re.codomain.ReverseIntensityContext;
+import omeis.providers.re.lut.LutProvider;
+import omeis.providers.re.quantum.QuantumFactory;
+import omero.model.Details;
+import omero.model.EventI;
+import omero.model.Experimenter;
+import omero.ApiUsageException;
+import omero.RString;
+import omero.ServerError;
+import omero.rtypes;
+import omero.api.IContainerPrx;
+import omero.api.IQueryPrx;
+import omero.api.RawPixelsStorePrx;
+import omero.api.ServiceFactoryPrx;
+import omero.model.IObject;
+import omero.model.WellSampleI;
+import omero.model.enums.UnitsLength;
+import omero.sys.ParametersI;
+import omero.util.IceMapper;
+import ome.units.UNITS;
+
+public class ImageDataRequestHandler {
+
+    private static final org.slf4j.Logger log =
+            LoggerFactory.getLogger(ImageDataRequestHandler.class);
+
+    private ImageDataCtx imageDataCtx;
+
+    private PixelsService pixelsService;
+
+    private List<Family> families;
+
+    List<RenderingModel> renderingModels;
+
+    LutProvider lutProvider;
+
+    private int init_zoom;
+
+    private boolean interpolate;
+
+    /**
+     * Mapper between <code>omero.model</code> client side Ice backed objects
+     * and <code>ome.model</code> server side Hibernate backed objects.
+     */
+    protected final IceMapper mapper = new IceMapper();
+
+    public ImageDataRequestHandler(ImageDataCtx imageDataCtx,
+            PixelsService pixelsService,
+            List<Family> families,
+            List<RenderingModel> renderingModels,
+            LutProvider lutProvider,
+            int init_zoom,
+            boolean interpolate) {
+        this.imageDataCtx = imageDataCtx;
+        this.pixelsService = pixelsService;
+        this.families = families;
+        this.renderingModels = renderingModels;
+        this.lutProvider = lutProvider;
+        this.init_zoom = init_zoom;
+        this.interpolate = interpolate;
+    }
+
+    public JsonObject getImageData(omero.client client) {
+        log.info("In ReqeustHandler::getImageData");
+        ServiceFactoryPrx sf = client.getSession();
+        try {
+            Long imageId = imageDataCtx.imageId;
+            IQueryPrx iQuery = sf.getQueryService();
+            ImageI image = queryImageData(iQuery, imageId);
+            if (image == null) {
+                return null;
+            }
+            Details details = image.getDetails();
+            Experimenter owner = details.getOwner();
+            EventI creationEvent = queryEvent(iQuery, details.getCreationEvent().getId().getValue());
```

Comment:
Maybe just expand the image data query to include the creation event so we don't have this second query?